tests/test_cases/test_array/test_array.py

```diff
@@ -292,6 +299,8 @@ async def test_discover_all(dut):
     elif LANGUAGE in ["verilog"] and cocotb.SIM_NAME.lower().startswith("riviera"):
         # Applies to Riviera-PRO 2019.10 and newer.
         pass_total = 180
+    elif LANGUAGE in ["verilog"] and "vcs" in SIM_NAME:
+        pass_total = 142
```

Comment:
What's missing? Only the signals in the generate loops, or something else?

Reply:
Looks like..  `asc_gen`, `desc_gen`  and all inside are missing.

-

Reply:
There is something more wrong (`OFFSET`  is inside generate) :
- VCS: https://github.com/cocotb/cocotb/actions/runs/12572898035/job/35045357869?pr=4324#step:24:1414
- Xcelium: https://github.com/cocotb/cocotb/actions/runs/12572898035/job/35045357716?pr=4324#step:24:1439

Reply:
VCS does not find `vpiGenScope` but it finds registers declared inside  with `vpiFullName` in form of  `top.foobar[0].gen_reg`

Reply:
https://github.com/cocotb/cocotb/issues/4328